agent/command.go

```diff
@@ -196,6 +196,11 @@ func (a *Agent) runCommandOrFunc(ctx context.Context, tc *taskContext, commandIn
 			commandSpan.End()
 			if cmd.RetryOnFailure() {
 				tc.logger.Task().Infof("Command is set to automatically restart on completion, this can be done %d total times per task.", evergreen.MaxAutomaticRestarts)
+				// Sometimes the command is cancelled due to timeouts but we still want the MarkFailedTaskToRestart to be called
+				// with a valid context.
+				if errors.Is(err, context.Canceled) {
+					ctx = context.WithoutCancel(ctx)
+				}
```

Comment:
I agree that it's a good idea to pass in a different context to MarkFailedTaskToRestart so that if the main context is canceled, it still runs. However, It don't think it should be wrapped in an errors check (it can be canceled after that check ) and it can be a standard context.Background().

Reply:
Sure, I'll just change it to always create a new context that gets cancelled at the end of the func